### DISCUSSION TITLE
importing from "jotai" and "jotai/react" is the same in v2? what about tree shaking?

### DISCUSSION THREAD

**Discussion by wootra** [pmndrs/jotai#1766] [Q&A] [answered]
I know jotai v2 is trying to support the other framework like vue. that's why there was some path changes like "jotai/react" or "jotai/vanilla". 
but in the v2 migration guide, the comment says they are the same as importing from "jotai". 
it can possibly lead misunderstanding that they two are actually the same. 
but I think when you bundle the package, they will pull the different amount of module. don't they? 
I don't know if there is some improvement on tree-shaking tech over time. 
But I want to know if they are actually the same.

**Comment by dai-shi** [comment, marked as answer]
Good question. I believe with ESM it's 100% the same. It can be tree-shaked. However, with CJS, it's not.

Let's see if there's any option in transpiler to improve. Otherwise, we can add a note about using `jotai/*` with CJS.

**Reply by dai-shi** [reply]
#1767

**Reply by wootra** [reply]
thanks for the answer! I didn't know it is automatically tree-shaking with ESM.
